packages/sdk/src/fee/FeeEstimator.ts

```diff
@@ -0,0 +1,164 @@
+import { Psbt } from "bitcoinjs-lib";
+import { Buffer } from "buffer";
+import { MAXIMUM_FEE } from "../constants";
+import { OrditSDKError } from "../errors";
+import { getNetwork, getScriptType } from "../utils";
+import type { AddressFormat } from "../addresses/types";
+import type { Network } from "../config/types";
+import type { FeeEstimatorOptions } from "./types";
+
+class FeeEstimator {
+  protected fee = 0;
+  protected feeRate: number;
+  protected network: Network;
+  protected psbt: Psbt;
+  protected witness?: Buffer[] = [];
+  protected virtualSize = 0;
+  protected weight = 0;
+
+  constructor({ feeRate, network, psbt, witness }: FeeEstimatorOptions) {
+    if (feeRate < 0 || !Number.isSafeInteger(feeRate)) {
+      throw new OrditSDKError("Invalid feeRate");
+    }
+
+    this.feeRate = +feeRate; // convert decimal to whole number that might have passed Number.isSafeInteger check due to precision loss
```

Comment:
@iamcrazycoder confused, do we disallow decimal fee rates here?

Reply:
No. Unary operator `+` parses the overflown number to a whole number. `+1.000000000000000001` would become `1` but it doesn't any difference since JS will always treat the former as a whole number.

You can remove the unary operator and the corresponding comment.